execution.go

```diff
@@ -764,6 +761,30 @@ func (e *QueryExecution) executeChildStep(ctx context.Context, step *queryPlanSt
 	}
 }
 
+// executeBrambleStep executes the Bramble-specific operations
+func (e *QueryExecution) executeBrambleStep(ctx context.Context, step *queryPlanStep, result map[string]interface{}) {
+	m := buildTypenameResponseMap(step.SelectionSet, step.ParentType)
+	mergeMaps(result, m)
+	e.wg.Done()
+}
+
+// buildTypenameResponseMap recursively builds the response map for `__typename`
+// fields. This is used for namespaces as they do not belong to any service.
+func buildTypenameResponseMap(ss ast.SelectionSet, currentType string) map[string]interface{} {
+	res := make(map[string]interface{})
+	for _, f := range selectionSetToFields(ss) {
+		if len(f.SelectionSet) > 0 {
+			res[f.Alias] = buildTypenameResponseMap(f.SelectionSet, f.Definition.Type.Name())
+			continue
+		}
+
+		if f.Name == "__typename" {
```

Comment:
```suggestion
		if f.Name == introspectionTypeNameField {
```
also `__schema` to `introspectionSchemaField`?

Reply:
I left this one as is because it's a standard field documented in the graphql spec. So the value shouldn't change and I think it makes the code easier to read. We also already use this kind of values elsewhere in the code.